### DISCUSSION TITLE
Help with null function

### DISCUSSION THREAD

**Discussion by slappycat2** [SilentVoid13/Templater#887] [Help] [answered]
Note: This was also posted in the Obsidian Git Discussions Tab:

Hello, and thanks for what has been up until this past week a great tool for Obsidian! I’ve been using it for about a month now and everything I’ve done with Obsidian and Templater has been fine. Lot’s of good examples, including [Obsidian-Scraper](https://github.com/basilioss/obsidian-scrapers) by [@Basilios](https://forum.obsidian.md/u/basilios) which I was able to modify to suit my needs, and it was all were working great up until about a week ago.
I’ve tried multiple clean installs of Obsidian and Templater, and no matter what, I get the same error. I think it all started with the Obsidian 1.0 upgrade.

### Things I have tried
I’ve upgraded to 1.0.3 and I have the latest version of Templater, but like I said, this started with 1.0. I’ve been working on debugging this, non-stop for the past week, searching issues, forums, etc. No luck, which I find really strange :-o!
I’m not a stack programmer, but I’m okay tweaking JS a little bit, as long as I can back it out if I need to.
For testing purposes I started with the first example given on the [Templater Introduction](https://silentvoid13.github.io/Templater/introduction.html) page, and I took out the daily quote, because I read a post where there was problems with that. But even just that bit of code fails.
I did have some success removing the front-matter, but after I tried to add it back, the error comes back, and then (and THIS IS REALLY WEIRD!), even if I remove it again, exit and come back in, I get the same error without the front-matter! It’s like something gets corrupted and it can’t be repaired.

### What I’m trying to do

What I’m trying to do is just create a new file with a template. For example just creating the daily note, with the above mentioned test template:

---
creation date: <% tp.file.creation_date() %>
modification date: <% tp.file.last_modified_date("dddd Do MMMM YYYY HH:mm:ss") %>
---

<< [[<% tp.date.now("YYYY-MM-DD", -1) %>]] | [[<% tp.date.now("YYYY-MM-DD", 1) %>]] >>

# <% tp.file.title %>

fails with the same error:

Templater Error: Template parsing error, aborting. ***
*** “” is not a function

I am about to give up at this point, so any help would be greatly appreciated!

This is the exact output from the console:

app.js:1 Obsidian Developer Console
plugin:dataview:20685 Dataview: version 0.5.47 (requires obsidian 0.13.11)
plugin:dataview:13951 Dataview: all 29 files have been indexed in 0.05s (29 cached, 0 skipped).
plugin:dataview:13955 Dataview: Dropped cache entries for 1 deleted files.
plugin:templater-obsidian:82 Templater Error: Template parsing error, aborting.
“” is not a function
log_error @ plugin:templater-obsidian:82
eval @ plugin:templater-obsidian:103
rejected @ plugin:templater-obsidian:57
Promise.then (async)
step @ plugin:templater-obsidian:62
eval @ plugin:templater-obsidian:63
__async @ plugin:templater-obsidian:47
errorWrapper @ plugin:templater-obsidian:98
eval @ plugin:templater-obsidian:3907
eval @ plugin:templater-obsidian:63
__async @ plugin:templater-obsidian:47
overwrite_file_commands @ plugin:templater-obsidian:3905
eval @ plugin:templater-obsidian:3995
fulfilled @ plugin:templater-obsidian:50
Promise.then (async)
step @ plugin:templater-obsidian:62
eval @ plugin:templater-obsidian:63
__async @ plugin:templater-obsidian:47
on_file_creation @ plugin:templater-obsidian:3973
eval @ plugin:templater-obsidian:4053
e.tryTrigger @ app.js:1
e.trigger @ app.js:1
t.trigger @ app.js:1
t.onChange @ app.js:1
e.trigger @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
m @ app.js:1
e.reconcileFileCreation @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
(anonymous) @ app.js:1
m @ app.js:1
e.reconcileFileInternal @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
(anonymous) @ app.js:1
m @ app.js:1
e.reconcileFile @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
m @ app.js:1
e.reconcileInternalFile @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
(anonymous) @ app.js:1
m @ app.js:1
(anonymous) @ app.js:1
n @ app.js:1
Promise.then (async)
e.queue @ app.js:1
e.write @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
(anonymous) @ app.js:1
m @ app.js:1
t.create @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
m @ app.js:1
e.createNewMarkdownFile @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
(anonymous) @ app.js:1
m @ app.js:1
e.getDailyNote @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
m @ app.js:1
e.onOpenDailyNote @ app.js:1

Again, any help would be greatly appreciated. TIA!

**Comment by AB1908** [comment]
Sorry to hear about your frustration! I'm more than happy to get on quick Discord call to narrow this down.

**Reply by AB1908** [reply]
Can you also post the template inside codeblocks? It seems some stuff got formatted out and it's making it harder to debug.

**Reply by slappycat2** [reply]
```

### from tpDocEG via Folder Link

<% tp.file.cursor(1) %>
```

**Reply by slappycat2** [reply]
To be honest, I've stripped it down to nothing and it will still fail with the same error.  It has to be something dumb I am doing. I am at a loss... Thanks AB1908!

**Comment by slappycat2** [comment]
Thanks, @AB1908 How do we get on quick discord. I'm on discord, but I don't see that. I will all post the template, if I can. TIA!

**Comment by slappycat2** [comment, marked as answer]
Thank you for responding, AB1908! @shabegom  was able to diagnose the issue
last night. It turns out Templater fails if even one script in the scripts
folder has an error in it, regardless of whether its being used or not.

Who knew!?!!

Regards,

-Swen
